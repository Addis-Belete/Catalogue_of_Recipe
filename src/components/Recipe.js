import React from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { fetchIngredients } from '../action/ingredient';
import styles from '../style/style.module.css';

const Recipe = ({ recipes }) => {
  const dispatch = useDispatch();
  const clickHandler = (id) => {
    dispatch(fetchIngredients(id));
  };

  return (
    <div>

      <div>
        {recipes.map((recipe) => (
          <div key={recipe.title}>
            <Link to={`/${recipe.id}`} onClick={clickHandler(recipe.id)}>
              <img className={styles.img} src={`${recipe.image}`} alt="recipeImage" />
              <p className={styles.p}>{recipe.title}</p>
            </Link>
          </div>
        ))}

      </div>

    </div>

  );
};
Recipe.propTypes = {
  recipes: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      title: PropTypes.string,
      image: PropTypes.string,
    }),
  ),
};
Recipe.defaultProps = {
  recipes: [],
};
export default Recipe;
