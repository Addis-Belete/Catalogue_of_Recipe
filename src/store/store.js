import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import { recipeReducer } from '../reducer';

const store = createStore(recipeReducer, composeWithDevTools(applyMiddleware(thunk)));
export default store;
