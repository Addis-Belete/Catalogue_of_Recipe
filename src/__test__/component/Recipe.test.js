import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Recipe from '../../components/Recipe';
import store from '../../store/store';

afterEach(cleanup);

it('should render the list of ingredients', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Recipe />
      </BrowserRouter>
    </Provider>,
  );
  expect(
    asFragment(
      <Provider store={store}>
        <Recipe />
      </Provider>,
    ),
  ).toMatchSnapshot();
});
