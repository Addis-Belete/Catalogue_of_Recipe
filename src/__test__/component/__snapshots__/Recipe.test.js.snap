// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the list of ingredients 1`] = `
<DocumentFragment>
  <div>
    <div
      class="tablet desktop"
    />
  </div>
</DocumentFragment>
`;
